src/main/java/bio/terra/service/auth/iam/IamResourceType.java

```diff
@@ -6,22 +6,30 @@
 import org.apache.commons.lang3.StringUtils;
 
 public enum IamResourceType {
-  DATAREPO("datarepo"),
-  DATASET("dataset"),
-  DATASNAPSHOT("datasnapshot"),
-  SPEND_PROFILE("spend-profile"),
-  WORKSPACE("workspace");
+  DATAREPO("datarepo", IamResourceTypeEnum.DATAREPO),
+  DATASET("dataset", IamResourceTypeEnum.DATASET),
+  DATASNAPSHOT("datasnapshot", IamResourceTypeEnum.DATASNAPSHOT),
+  SPEND_PROFILE("spend-profile", IamResourceTypeEnum.SPEND_PROFILE),
+  SNAPSHOT_BUILDER_REQUEST(
+      "snapshot-builder-request", IamResourceTypeEnum.SNAPSHOT_BUILDER_REQUEST),
+  WORKSPACE("workspace", IamResourceTypeEnum.WORKSPACE);
 
   private final String samResourceName;
+  private final IamResourceTypeEnum iamResourceTypeEnum;
 
-  IamResourceType(String samResourceName) {
+  IamResourceType(String samResourceName, IamResourceTypeEnum iamResourceTypeEnum) {
     this.samResourceName = samResourceName;
+    this.iamResourceTypeEnum = iamResourceTypeEnum;
   }
 
   public String getSamResourceName() {
     return samResourceName;
   }
 
+  public IamResourceTypeEnum getIamResourceTypeEnum() {
```

Comment:
If this value is unused outside this class, then it doesn't need a getter. Since enums can't be subclassed it's OK to refer to the field directly within the class too.